src/beacon.js

```diff
@@ -91,7 +95,8 @@ export default function Emitter(endpoint, context, options) {
         editedMessage = message.payload || {};
         editedMessage.type = message.type;
 
-        if (!send(endpoint, JSON.stringify(editedMessage), sync)) {
+        if (!send(endpoint, JSON.stringify(editedMessage), sync) && RETRIES.current) {
+          --RETRIES.current
```

Comment:
This will have concurrency issues won't it? You can call emit repeatedly and that would reset the counter -- can we pass which attempt number it is into the the transmit call?

Reply:
If emit is getting called, it means we are sending new messages and it's I'm resetting the retries count. We try the new message up to 10 and if it failed, it stops more tries. In a case that emit is getting called continuously, that's another loop we need to fix.